app/src/main/java/io/zenandroid/onlinego/ui/screens/game/GameViewModel.kt

```diff
@@ -288,6 +294,7 @@ class GameViewModel(
             } ?: game?.moves?.size ?: 0
             val nextButton = Next(analysisShownMoveNumber < maxAnalysisMoveNumber)
             val chatButton = Chat(if(unreadMessagesCount > 0) unreadMessagesCount.toString() else null)
+            val pauseButton = if (game?.pauseControl.isPlayerPaused()) Resume else Pause
```

Comment:
Yes, you are correct and the issue is down to the intricacies of using Jake Wharton's Molecule library for state management. Basically, everything in the big lambda that is called with `moleculeScope.launchMolecule` is Compose code (yes, Compose code that produces state instead of producing UI!), and as such, it needs to have a guarantee of stability OR use a snapshot state (e.g. `mutableStateOf`). This is the reason for the 30-or-so mutable states at the beginning of the ViewModel. What is happening right now, is that this line is run only once since nothing is notifying the Compose Compiler that the actual value changed.

The correct solution here is to create a new snapshot state in the ViewModel (e.g. `private var gamePaused by mutableStateOf(false)`), update that in `onGameChanged` (e.g. `gamePaused = game?.pauseControl.isPlayerPaused()`) and then at this line use `if(gamePaused) ...`